src/infrastructure/repository/article/ArticleRepository.ts

```diff
@@ -46,21 +49,27 @@ export class ArticleRepository implements IArticleRepository {
     )
   }
 
-  public async findAll(query: QueryParameters) {
+  public async findAll(articles: Articles) {
     const connection = await mysql.createConnection(config.db).catch((error) => {
       console.error(error)
       throw new DataBaseError(HTTP_ERROR_MESSAGE.DataBaseConnectionError)
     })
-    const sql = `SELECT * FROM ${this.table} LIMIT ? OFFSET ?`
+    const articleGetSql = `SELECT * FROM ${this.table} LIMIT ? OFFSET ?`
     const [result] = await connection
-      .execute<responseJson[]>(sql, [query.limit, query.offset])
+      .execute<responseJson[]>(articleGetSql, [articles.limit, articles.offset])
       .catch((error) => {
         console.error(error)
         throw new DataBaseError(HTTP_ERROR_MESSAGE.DataBaseQueryError)
       })
     if (this.notFindData(result)) {
-      return null
+      return new Articles([], 0)
     }
+    const totalArticleCountSql = `SELECT COUNT(*)as total FROM ${this.table}`
```

Comment:
細かいところですがここだけ直しておいてください！
```suggestion
    const totalArticleCountSql = `SELECT COUNT(*) AS total FROM ${this.table}`
```